src/net_processing.cpp

```diff
@@ -4200,6 +4209,53 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
         peer->m_recon_state->m_next_recon_respond = NextReconRespond(current_time);
         return;
     }
+
+    // Received a response to the reconciliation request.
+    // May leak tx-related privacy if we announce local transactions right away, if a peer is strategic about sending
+    // sketches to us via different connections (requires attacker to occupy multiple outgoing connections).
+    if (msg_type == NetMsgType::SKETCH) {
+        if (peer->m_recon_state == nullptr) return;
+        if (peer->m_recon_state->m_outgoing_recon != Peer::ReconState::ReconPhase::INIT_REQUESTED) return;
+
+        std::vector<uint8_t> skdata;
+        vRecv >> skdata;
+
+        // Attempt to decode the received sketch with a local sketch.
+        if (skdata.size() / BYTES_PER_SKETCH_CAPACITY > MAX_SKETCH_CAPACITY) return;
+        uint16_t remote_sketch_capacity = uint16_t(skdata.size() / BYTES_PER_SKETCH_CAPACITY);
+        minisketch* remote_sketch = minisketch_create(RECON_FIELD_SIZE, 0, remote_sketch_capacity);
+        uint8_t remote_sketch_serialized[MAX_SKETCH_CAPACITY * BYTES_PER_SKETCH_CAPACITY];
+        std::copy(skdata.begin(), skdata.end(), remote_sketch_serialized);
+        minisketch_deserialize(remote_sketch, remote_sketch_serialized);
+
+        minisketch* working_sketch; // Contains sketch of the set difference
+        minisketch* local_sketch = peer->m_recon_state->ComputeSketch(peer->m_recon_state->m_local_set, remote_sketch_capacity);
```

Comment:
I think we need to call `minisketch_destroy` on these somewhere to avoid a memory leak right?